src/anemoi/transform/filters/lambda_filters.py

```diff
@@ -0,0 +1,94 @@
+import typing as tp
+import importlib
+
+from anemoi.transform.filters.base import SimpleFilter
+from anemoi.transform.filters import filter_registry
+from earthkit.data.core.fieldlist import Field, FieldList
+
+
+
+class EarthkitFieldLambdaFilter(SimpleFilter):
+    """A filter to apply an arbitrary function to individual fields.
+
+    Parameters
+    ----------
+    fn: callable or str
+        The lambda function as a callable with the general signature 
+        `fn(*earthkit.data.Field, *args, **kwargs) -> earthkit.data.Field` or
+        a string path to the function, such as "package.module.function".
+    param: list or str
+        The parameter name or list of parameter names to apply the function to.
+    args: list
+        The list of arguments to pass to the lambda function.
+    kwargs: dict
+        The dictionary of keyword arguments to pass to the lambda function.
+    backward_fn (optional): callable, str or None
+        The backward lambda function as a callable with the general signature 
+        `backward_fn(*earthkit.data.Field, *args, **kwargs) -> earthkit.data.Field` or
+        a string path to the function, such as "package.module.function".
+
+    """
+
+    def __init__(
+        self,
+        fn: str | tp.Callable[[Field, tp.Any], Field],
+        param: str | list[str],
+        args: list = [],
+        kwargs: dict[str, tp.Any] = {},
+        backward_fn: str | tp.Callable[[Field, tp.Any], Field] | None = None,
+    ):  
+        
+        if not isinstance(args, list):
+            raise ValueError("Expected 'args' to be a list. "
+                             f"Got {args} instead.")
+        if not isinstance(kwargs, dict):
+            raise ValueError("Expected 'kwargs' to be a dictionary. "
+                             f"Got {kwargs} instead.")
+
+        self.fn = self._import_fn(fn) if isinstance(fn, str) else fn
+
+        if backward_fn:
+            self.backward_fn = self._import_fn(backward_fn) \
+                if isinstance(backward_fn, str) else backward_fn
```

Comment:
The attribute is assigned in any case (if a string, it is first converted to a function, otherwise it is assigned directly, which includes being a function or None) so it shouldn't result in an `AttributeError`.